modules/api/src/main/SocketTestResult.scala

```diff
@@ -0,0 +1,24 @@
+package lila.api
+
+import play.api.libs.json.*
+
+import lila.db.JSON
+import lila.core.config.NetConfig
+
+final class SocketTestResult(resultsDb: lila.db.AsyncCollFailingSilently)(using Executor):
+  def put(results: JsObject) = resultsDb: coll =>
+    coll.insert.one(JSON.bdoc(results)).void
+
+object SocketTest:
+
+  def isUserInTestBucket(net: NetConfig)(using ctx: Context) =
+    net.socketTest && ctx.pref.usingAltSocket.isEmpty && ctx.userId.exists: userId =>
+      java.security.MessageDigest.getInstance("MD5").digest(userId.value.getBytes).head % 128 == 0
```

Comment:
how about `ctx.me.exists(_.createdAt.getNano % 128 == 0)`, that looks cheaper

Reply:
as long as there is a lila endpoint or dev cli command to check manually (for support). i'm not super attached and saving cycles spent repeatedly xoring numbers seems good, but on the other hand inclusion in the md5 test bucket can be determined with with no reliance on prod mongodb.

Reply:
there's always `username.toList.map(_.toInt).sum % 128`

Reply:
seems fine. i'm not sure we need the avalanche effect of a strong hash. even if the modulo sum of ascii characters results in uneven buckets with weird correlations, it's not a huge deal.

Reply:
just don't tell niklas or glbert.